src/bun.js/bindings/VMModuleScript.cpp

```diff
@@ -0,0 +1,277 @@
+    JSValue contextArg = args.at(0);
+    bool isContext;
+    if (!contextArg.isObject()) {
+        isContext = false;
+    } else {
+        auto* zigGlobalObject = reinterpret_cast<Zig::GlobalObject*>(globalObject);
+        isContext = zigGlobalObject->vmModuleContextMap()->has(asObject(contextArg));
+    }
+    return JSValue::encode(jsBoolean(isContext));
+}
+
+class VMModuleScriptPrototype final : public JSNonFinalObject {
+public:
+    using Base = JSNonFinalObject;
+
+    static VMModuleScriptPrototype* create(VM& vm, JSGlobalObject* globalObject, Structure* structure)
+    {
+        VMModuleScriptPrototype* ptr = new (NotNull, allocateCell<VMModuleScriptPrototype>(vm)) VMModuleScriptPrototype(vm, structure);
+        ptr->finishCreation(vm);
+        return ptr;
+    }
+
+    DECLARE_INFO;
+    template<typename CellType, SubspaceAccess>
+    static GCClient::IsoSubspace* subspaceFor(VM& vm)
+    {
+        return &vm.plainObjectSpace();
+    }
+    static Structure* createStructure(VM& vm, JSGlobalObject* globalObject, JSValue prototype)
+    {
+        return Structure::create(vm, globalObject, prototype, TypeInfo(ObjectType, StructureFlags), info());
+    }
+
+private:
+    VMModuleScriptPrototype(VM& vm, Structure* structure)
+        : Base(vm, structure)
+    {
+    }
+
+    void finishCreation(VM&);
+};
+STATIC_ASSERT_ISO_SUBSPACE_SHARABLE(VMModuleScriptPrototype, VMModuleScriptPrototype::Base);
+
+static const struct HashTableValue scriptPrototypeTableValues[] = {
+   { "cachedDataRejected"_s, static_cast<unsigned>(PropertyAttribute::ReadOnly|PropertyAttribute::CustomAccessor), NoIntrinsic, { HashTableValue::GetterSetterType, scriptGetCachedDataRejected, nullptr } },
+   { "createCachedData"_s, static_cast<unsigned>(PropertyAttribute::ReadOnly|PropertyAttribute::Function), NoIntrinsic, { HashTableValue::NativeFunctionType, scriptCreateCachedData, 0 } },
+   { "runInContext"_s, static_cast<unsigned>(PropertyAttribute::ReadOnly|PropertyAttribute::Function), NoIntrinsic, { HashTableValue::NativeFunctionType, scriptRunInContext, 0 } },
+   { "runInThisContext"_s, static_cast<unsigned>(PropertyAttribute::ReadOnly|PropertyAttribute::Function), NoIntrinsic, { HashTableValue::NativeFunctionType, scriptRunInThisContext, 0 } },
+   { "sourceMapURL"_s, static_cast<unsigned>(PropertyAttribute::ReadOnly|PropertyAttribute::CustomAccessor), NoIntrinsic, { HashTableValue::GetterSetterType, scriptGetSourceMapURL, nullptr } },
+};
+
+const ClassInfo VMModuleScriptPrototype::s_info = { "Script"_s, &Base::s_info, nullptr, nullptr, CREATE_METHOD_TABLE(VMModuleScriptPrototype) };
+const ClassInfo VMModuleScript::s_info = { "Script"_s, &Base::s_info, nullptr, nullptr, CREATE_METHOD_TABLE(VMModuleScript) };
+const ClassInfo VMModuleScriptConstructor::s_info = { "Script"_s, &Base::s_info, nullptr, nullptr, CREATE_METHOD_TABLE(VMModuleScriptConstructor) };
+
+VMModuleScriptConstructor::VMModuleScriptConstructor(VM& vm, Structure* structure)
```

Comment:
Maybe `NodeVMScript` would be a better name? `ModuleScript` might be confused for ECMA Module